server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/MirrorConfig.java

```diff
@@ -99,6 +100,11 @@ public MirrorConfig(@JsonProperty("id") String id,
         this.localRepo = requireNonNull(localRepo, "localRepo");
         this.localPath = firstNonNull(localPath, "/");
         this.remoteUri = requireNonNull(remoteUri, "remoteUri");
+
+        // Validate the remote URI.
+        final String suffix = remoteUri.getScheme().equals(SCHEME_DOGMA) ? "dogma" : "git";
```

Comment:
Shouldn't we do this in the `MirrorDto` or `MirroringServiceV1`?
How about also checking if the local repo exists?

Reply:
> Shouldn't we do this in the MirrorDto or MirroringServiceV1?

I chose this place because the segments of remote URI in `MirrorDto` need to be assembled.
https://github.com/line/centraldogma/blob/5fec9f111c862f47d24350a1d16f8c36b8acf0a8/server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/DefaultMetaRepository.java#L301-L306

> How about also checking if the local repo exists?

That would be covered by Mirror UI which provides a select box to list local repos.
However, validation on the server side is a also good idea. I will do it in the f/u work. 😉